backend/services/inventory/api/http/api_inventory.go

```diff
@@ -564,286 +452,292 @@ func (i *inventoryHandlers) UpdateDeviceTagsHandler(w rest.ResponseWriter, r *re
-	w.WriteHeader(http.StatusNoContent)
+	c.Status(http.StatusNoContent)
 }
 
-func (i *inventoryHandlers) AddDeviceToGroupHandler(w rest.ResponseWriter, r *rest.Request) {
-	ctx := r.Context()
-
-	l := log.FromContext(ctx)
+func (i *ManagementAPI) AddDeviceToGroupHandler(c *gin.Context) {
+	ctx := c.Request.Context()
 
-	devId := r.PathParam("id")
+	devId := c.Param("id")
 
 	var group InventoryApiGroup
-	err := r.DecodeJsonPayload(&group)
+	err := c.ShouldBindJSON(&group)
 	if err != nil {
-		u.RestErrWithLog(
-			w, r, l, errors.Wrap(err, "failed to decode device group data"),
-			http.StatusBadRequest)
+		rest.RenderError(c,
+			http.StatusBadRequest,
+			errors.Wrap(err, "failed to decode device group data"))
 		return
 	}
 
 	if err = group.Validate(); err != nil {
-		u.RestErrWithLog(w, r, l, err, http.StatusBadRequest)
+		rest.RenderError(c, http.StatusBadRequest, err)
 		return
 	}
 
-	err = i.inventory.UpdateDeviceGroup(ctx, model.DeviceID(devId), model.GroupName(group.Group))
+	err = i.App.UpdateDeviceGroup(ctx, model.DeviceID(devId), model.GroupName(group.Group))
 	if err != nil {
 		if cause := errors.Cause(err); cause != nil && cause == store.ErrDevNotFound {
-			u.RestErrWithLog(w, r, l, err, http.StatusNotFound)
+			rest.RenderError(c, http.StatusNotFound, err)
 			return
 		}
-		u.RestErrWithLogInternal(w, r, l, err)
+		rest.RenderInternalError(c, err)
 		return
 	}
-	w.WriteHeader(http.StatusNoContent)
+	c.Status(http.StatusNoContent)
 }
 
-func (i *inventoryHandlers) GetDevicesByGroupHandler(w rest.ResponseWriter, r *rest.Request) {
-	ctx := r.Context()
+func (i *ManagementAPI) GetDevicesByGroupHandler(c *gin.Context) {
+	ctx := c.Request.Context()
 
-	l := log.FromContext(ctx)
+	group := c.Param("name")
 
-	group := r.PathParam("name")
-
-	page, perPage, err := utils.ParsePagination(r)
+	page, perPage, err := rest.ParsePagingParameters(c.Request)
 	if err != nil {
-		u.RestErrWithLog(w, r, l, err, http.StatusBadRequest)
+		rest.RenderError(c, http.StatusBadRequest, err)
 		return
 	}
 
 	//get one extra device to see if there's a 'next' page
-	ids, totalCount, err := i.inventory.ListDevicesByGroup(
+	ids, totalCount, err := i.App.ListDevicesByGroup(
 		ctx,
 		model.GroupName(group),
 		int((page-1)*perPage),
 		int(perPage),
 	)
 	if err != nil {
 		if err == store.ErrGroupNotFound {
-			u.RestErrWithLog(w, r, l, err, http.StatusNotFound)
+			rest.RenderError(c, http.StatusNotFound, err)
+
 		} else {
-			u.RestErrWithLogInternal(w, r, l, err)
+			rest.RenderError(c,
+				http.StatusInternalServerError,
+				errors.New("internal error"),
+			)
 		}
 		return
 	}
 
 	hasNext := totalCount > int(page*perPage)
 
-	links := utils.MakePageLinkHdrs(r, page, perPage, hasNext)
+	hints := rest.NewPagingHints().
+		SetPage(page).
+		SetPerPage(perPage).
+		SetHasNext(hasNext).
+		SetTotalCount(int64(totalCount))
+
+	links, err := rest.MakePagingHeaders(c.Request, hints)
+	if err != nil {
+		rest.RenderInternalError(c, err)
+		return
+	}
```

Comment:
In the (impossible) event that this function returns an error it should be because of a client error. However, since the PerPage and Page parameters are set in the hints, I don't believe the above call can return an error :shrug:
```suggestion
	if err != nil {
		rest.RenderError(c, http.StatusBadRequest, err)
		return
	}
```